### PR TITLE
Added types for @americanexpress/react-seo

### PR DESCRIPTION
**Pull request by Zenoo**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/americanexpress__react-seo/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    description: string;
+    canonical?: string;
+    image?: Media;
+    video?: Media;
+    openGraph?: OpenGraph;
+    twitterCard?: TwitterCard;
+    keywords?: string[];
+    locale?: string;
+    meta?: object[];
+    siteUrl?: string;
```
Comment (peterblazejewicz) [inline types/americanexpress__react-seo/index.d.ts:76]:
```suggestion
    siteUrl: string;
```
is this one required?
https://github.com/americanexpress/react-seo/blob/4ea08150715a24a18681c8d897912bac05e0b788/src/components/SEO.jsx#L104

File: types/americanexpress__react-seo/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    site?: string;
+    siteId?: string;
+    creator?: string;
+    creatorId?: string;
+    app?: TwitterApp;
+    player?: TwitterPlayer;
+}
+
+export interface SEOProps {
+    title?: string;
```
Comment (peterblazejewicz) [inline types/americanexpress__react-seo/index.d.ts:66]: is this one required?
```suggestion
    title: string;
```
https://github.com/americanexpress/react-seo/blob/4ea08150715a24a18681c8d897912bac05e0b788/src/components/SEO.jsx#L105

**Comment by peterblazejewicz** [comment]
please consider supporting TS 4.4 exact optional property types feature (`prop?: string | undefined`) for better interop with other libs or client code setup with strict features enabled:
https://www.typescriptlang.org/play?ts=4.4.2&exactOptionalPropertyTypes=true#example/exact-optional-properties

**Comment by Zenoo** [comment]
@peterblazejewicz So I should add `| undefined` to every optional prop, right?

**Comment by peterblazejewicz** [comment]
> So I should add `| undefined` to every optional prop, right?

not literally, but usually all are applicable. You can skip this one, as this is optional, review this later if needed in future

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@Zenoo thx!

**Comment by Zenoo** [comment]
Ready to merge

**Pull request merged**